multicluster/cmd/link.go

```diff
@@ -322,6 +327,7 @@ A full list of configurable values can be found at https://github.com/linkerd/li
 	cmd.Flags().StringVar(&opts.dockerRegistry, "registry", opts.dockerRegistry, "Docker registry to pull service mirror controller image from")
 	cmd.Flags().StringVarP(&opts.selector, "selector", "l", opts.selector, "Selector (label query) to filter which services in the target cluster to mirror")
 	cmd.Flags().StringVar(&opts.gatewayAddresses, "gateway-addresses", opts.gatewayAddresses, "If specified overwrites gateway addresses when gateway service is not type LoadBalancer (comma separated list)")
+	cmd.Flags().Uint32Var(&opts.gatewayPort, "gateway-port", opts.gatewayPort, "If specified, overwrites gateway port when gateway service is not type LoadBalancer")
```

Comment:
Based on the description, does this mean that the gateway port shouldn't be configurable if the gateway service is of type `LoadBalancer`? Is this something we need to enforce in the code? What would happen if we configure a overwrite the port in this scenario?

Reply:
That is a good question. It is mainly copied from the statement above.

As far as I can see, we don't do specific filtering on types; and if for any reason a service doesn't know its public ports, this can be useful. That said; I think that is in general only relevant for ClusterIP or NodePort services.